src/core/bluetoothdevicemodel.h

```diff
@@ -0,0 +1,92 @@
+/***************************************************************************
+  bluetoothdevicemodel.h - BluetoothDeviceModel
+
+ ---------------------
+ begin                : 20.11.2020
+ copyright            : (C) 2020 by David Signer
+ email                : david (at) opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef BLUETOOTHDEVICEMODEL_H
+#define BLUETOOTHDEVICEMODEL_H
+
+#include <QAbstractListModel>
+#include <QtBluetooth/QBluetoothServiceDiscoveryAgent>
+#include <QtBluetooth/QBluetoothServiceInfo>
+#include <QtBluetooth/QBluetoothLocalDevice>
+
+/**
+ * A model that provides all paired bluetooth devices name/address that are accessible over the serial port
+ */
+class BluetoothDeviceModel : public QAbstractListModel
+{
+    Q_OBJECT
+
+    Q_PROPERTY( ScanningStatus scanningStatus READ scanningStatus WRITE setScanningStatus NOTIFY scanningStatusChanged )
+    Q_PROPERTY( QString lastError READ lastError WRITE setLastError NOTIFY lastErrorChanged )
+
+    Q_ENUMS( BluetoothDeviceRoles )
+
+  public:
+    //! The roles provided by this model
+    enum BluetoothDeviceRoles
+    {
+      DeviceAddressRole = Qt::UserRole + 1,
+      DisplayStringRole
+    };
+
+    enum ScanningStatus
+    {
+      Scanning,
+      Succeeded,
+      Failed,
+      Canceled,
+      NoStatus
+    };
+    Q_ENUM( ScanningStatus )
+
+    /**
+     * Create a new value map model
```

Comment:
Leftover?
IMO you can completely skip constructor docstrings (unless they have more to say then the class docstring)